src/verify_cert.rs

```diff
@@ -745,122 +453,58 @@ where
-        #[cfg(feature = "alloc")]
-        {
-            let err = result.unwrap_err();
-            println!("{:?}", err.clone());
-        }
-
-        // It should be possible to build a revocation options builder with defaults.
-        let crl = include_bytes!("../tests/crls/crl.valid.der");
-        let crl =
-            &BorrowedCertRevocationList::from_der(&crl[..]).unwrap() as &dyn CertRevocationList;
-        let crls = [crl];
-        let builder = RevocationOptionsBuilder::new(&crls[..]).unwrap();
-        #[cfg(feature = "alloc")]
-        {
-            // The builder should be debug, and clone when alloc is enabled
-            println!("{:?}", builder);
-            _ = builder.clone();
-        }
-        let opts = builder.build();
-        assert_eq!(opts.depth, RevocationCheckDepth::Chain);
-        assert_eq!(opts.status_requirement, UnknownStatusPolicy::Deny);
-        assert_eq!(opts.crls.len(), 1);
-
-        // It should be possible to build a revocation options builder with custom depth.
-        let opts = RevocationOptionsBuilder::new(&crls[..])
-            .unwrap()
-            .with_depth(RevocationCheckDepth::EndEntity)
-            .build();
-        assert_eq!(opts.depth, RevocationCheckDepth::EndEntity);
-        assert_eq!(opts.status_requirement, UnknownStatusPolicy::Deny);
-        assert_eq!(opts.crls.len(), 1);
-
-        // It should be possible to build a revocation options builder that allows unknown
-        // revocation status.
-        let opts = RevocationOptionsBuilder::new(&crls[..])
-            .unwrap()
-            .allow_unknown_status()
-            .build();
-        assert_eq!(opts.depth, RevocationCheckDepth::Chain);
-        assert_eq!(opts.status_requirement, UnknownStatusPolicy::Allow);
-        assert_eq!(opts.crls.len(), 1);
-
-        // It should be possible to specify both depth and unknown status requirements together.
-        let opts = RevocationOptionsBuilder::new(&crls[..])
-            .unwrap()
-            .allow_unknown_status()
-            .with_depth(RevocationCheckDepth::EndEntity)
-            .build();
-        assert_eq!(opts.depth, RevocationCheckDepth::EndEntity);
-        assert_eq!(opts.status_requirement, UnknownStatusPolicy::Allow);
-        assert_eq!(opts.crls.len(), 1);
-
-        // The same should be true for explicitly forbidding unknown status.
-        let opts = RevocationOptionsBuilder::new(&crls[..])
-            .unwrap()
-            .forbid_unknown_status()
-            .with_depth(RevocationCheckDepth::EndEntity)
-            .build();
-        assert_eq!(opts.depth, RevocationCheckDepth::EndEntity);
-        assert_eq!(opts.status_requirement, UnknownStatusPolicy::Deny);
-        assert_eq!(opts.crls.len(), 1);
-
-        // Built revocation options should be debug and clone when alloc is enabled.
-        #[cfg(feature = "alloc")]
-        {
-            println!("{:?}", opts.clone());
+    fn role(&self) -> Role {
```

Comment:
I think this would be more ergonomic as a `From` impl on `Role` for `&PathNode` like:

```rust
impl From<&PathNode<'_>> for Role {
    fn from(node: &PathNode<'_>) -> Self {
        match node.issued {
            None => Self::EndEntity,
            Some(_) => Self::Issuer,
        }
    }
}
```

The one call-site on L50 can become `let role = path.into();` in that case.

Reply:
I quite like the explicitness and clarity of `node.role()`.

I think `into()` is best used where the resulting type is an obvious (so the reader doesn't need to do type inference in their head to work out what it means) and permanent (usually `into()` means the receiver stops existing, and the docs say "A value-to-value conversion that consumes the input value", though obviously defining it on a reference type makes that less interesting.)

Reply:
That's fair, I don't feel strongly :+1:

Reply:
Yes, I agree with @ctz here. The way I would explain it is that `impl From<&PathNode<'_>> for Role` doesn't hold up as a conceptual relationship even if the types work out okay.